flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaWriterITCase.java

```diff
@@ -401,16 +401,16 @@ void usePoolForTransactional() throws Exception {
             writer.snapshotState(1);
             assertThat(committables0).hasSize(1);
             final KafkaCommittable committable = committables0.stream().findFirst().get();
-            assertThat(committable.getProducer().isPresent()).isTrue();
+            assertThat(committable.getProducer()).isPresent();
 
             FlinkKafkaInternalProducer<?, ?> firstProducer =
                     committable.getProducer().get().getObject();
             assertThat(firstProducer != writer.getCurrentProducer())
                     .as("Expected different producer")
-                    .isTrue();
+                    .isNotSameAs(writer.getCurrentProducer());
```

Comment:
seems like a wrong statement which always will pass the test since `firstProducer != writer.getCurrentProducer()` is always is not same as `writer.getCurrentProducer()`

I guess the correct one should be 
```java
assertThat(firstProducer)
                    .as("Expected different producer")
                    .isNotSameAs(writer.getCurrentProducer());
```